aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -326,17 +328,60 @@ Tensor imag(const Tensor& self) {
   }
 }
 
-Tensor& conj_out(const Tensor& self, Tensor& result) {
-  return unary_op_impl_out(result, self, conj_stub);
+Tensor& conj_physical_out(const Tensor& self, Tensor& result) {
+  return unary_op_impl_out(result, self, conj_physical_stub);
+}
+
+Tensor _conj_physical(const Tensor& self) {
+  if (self.is_conj()) {
+    return self.conj().clone();
+  }
+  return unary_op_impl(self, at::conj_physical_out);
+}
+
+Tensor conj_physical(const Tensor& self) {
```

Comment:
I'm looking at `conj_physical` and `conj_physical_`. Suppose I start with tensor `x` with conjugate bit set. `x.conj_physical()` will return a tensor with the same data as x and conjugate bit unset: `x.conj().clone()`. `x.conj_physical_()` will conjugate `x` data in physical memory and will preserve conjugate bit. To me, this is confusing behavior, usually results of inplace and functional operations are the same, except for a few special cases.

Reply:
> I'm looking at `conj_physical` and `conj_physical_`. Suppose I start with tensor `x` with conjugate bit set. `x.conj_physical()` will return a tensor with the same data as x and conjugate bit unset: `x.conj().clone()`. `x.conj_physical_()` will conjugate `x` data in physical memory and will preserve conjugate bit. To me, this is confusing behavior, usually results of inplace and functional operations are the same, except for a few special cases.

Mmm I understand however, it is important to conserve the conjugate bit in in-place operations since the tensor might have other views. This PR tries to keep the behavior consistent within each category of operations: functional, in-place, and out-of-place operations, hence we materialize the conjugation before performing `conj_physical()`.

Reply:
This has nothing to do with `conj_physical` because all other inplace-functional operators have the same semantics. If I have x with conjugate bit, `x.add(2)` will give me a tensor with conjugate bit unset (having implicitly done the conjugation and then adding 2), and `x.add_(2)` will preserve conjugate bit (and add two).

Reply:
If we say "conj bit is like strides" it should be propagated like strides, and not be unset at the earliest possible moment?

Reply:
synced offline: for now, we only propagate the conj bit for inplace and view operations but might change this in future if there's significant upside.